### PR TITLE
Move post tag state to the post form

### PR DESCRIPTION
**Pull request by mgjeffries**
# Description
We wanted to use the power of django for good! Simplifying the client side code is good.
Also, the new wireframe requires us to list the post tags on the post form.
This pr cuts out the obsolete notion of post tag state, and instead relies on the server to manage the many-to-many relationship between posts and tags. The post tag form is cut down to a simple conditional rendering component, that is managed by the post form.
## Type of change
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
# How Has This Been Tested?
Please describe step-by-step the process that a reviewer can follow to adequately test this PR. 
- [x] Verify that post tags are still displayed on the detail page of posts
- [x] Verify that when a user view the new post form, a list of post tags is displayed
- [x] Verify that when a user submits a post with tags, the tags are sent to the server as a list on the post object
- [x] Verify that when a user views a post that has already been created, the post tags that have already been selected are still selected
- [x] Note that editing a post is not yet supported on the server side.
# Checklist:
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings

### CODE REVIEW

**Review by skratz17 — APPROVED** [review]
love it

**Review by FORDBA — APPROVED** [review]
Looks good

File: src/components/posts/postForm.js [hunk 1]
```diff
@@ -34,10 +36,28 @@ export const PostForm = (props) => {
       imageRef.current.value = post.image_url
       contentRef.current.value = post.content
 
-
```
Comment (gavinschriver) [inline src/components/posts/postForm.js:37]: @mgjeffries HEY this is already approved BUT - if you fix this ONE line in the edit mode field assignments it works PERFECT with the new server update branch im going to PR
postForm.js @ line 35 change to 
` categoryRef.current.value = post.category.id` (this is how the category id is accessible in the current representation of a post)

Comment (mgjeffries) [reply src/components/posts/postForm.js:37]: Will do!

**Pull request merged**